src/shared-components/LoadingState/LoadingState.tsx

```
@@ -115,6 +116,20 @@ const LoadingState = ({
     );
   }
 
+  // Custom variant: renders user-provided custom loader
+  if (variant === 'custom') {
+    return (
+      <div
+        data-testid={dataTestId}
+        role="status"
```

Biome (2.1.2)

[error] The elements with this role can be changed to the following elements:
<output>

For examples and more information, see WAI-ARIA Roles

(lint/a11y/useSemanticElements)